### PR TITLE
store cluster domain in workspace_table

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/workspaces/views.py [hunk 1]
```diff
@@ -54,8 +55,10 @@ def post(self, request):
             workspace.user.add(User.objects.get(user_id=request.user))
             cache.delete(str(workspace.id))
         else:
-            workspace = Workspace.objects.create(name=org_name, fyle_org_id=org_id)
+            fyle_connector = FyleConnector(auth_tokens.refresh_token)
+            cluster_domain = fyle_connector.get_cluster_domain()['cluster_domain']
```
Comment (ashwin1111) [inline apps/workspaces/views.py:59]: Please check app as well, we have a /cluster_domain api which returns it. You'll have to take care of all that. Basically only 1 call per workspace at time of creation should go to fyle. Rest should read from table.

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: apps/workspaces/views.py (hunk 1 above)
Comment (NileshPant1999) [reply apps/workspaces/views.py:59]: okay will take care of this.

**Review by ashwin1111 — APPROVED** [review]

**Comment by ashwin1111** [comment]
TODO: Have scripts ready before deploy

**Review by Sravanksk — APPROVED** [review]

File: apps/sage_intacct/utils.py [hunk 2]
```diff
@@ -485,13 +485,12 @@ def get_expense_link(self, lineitem) -> str:
         :param expense: Expense Lineitem
         :return: Expense link
         """
-        fyle_credentials = FyleCredential.objects.get(workspace_id=self.workspace_id)
-        fyle_connector = FyleConnector(fyle_credentials.refresh_token, self.workspace_id)
-        org_id = Workspace.objects.get(id=self.workspace_id).fyle_org_id
+        workspace = Workspace.objects.get(id=self.workspace_id)
+        org_id = workspace.fyle_org_id
+        cluster_domain = workspace.cluster_domain
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:490]: sir, this approach would break for existing clients

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: apps/sage_intacct/utils.py (hunk 2 above)
Comment (Shwetabhk) [reply apps/sage_intacct/utils.py:490]:
```python
if not cluster_domain:
   fyle_credentials = FyleCredential.objects.get(workspace_id=self.workspace_id)
   fyle_connector = FyleConnector(fyle_credentials.refresh_token, self.workspace_id)
   cluster_domain = fyle_connector.get_cluster_domain()
   workspace.cluster_domain = cluster_domain
   workspace.save()
```

**Pull request merged**